### DISCUSSION TITLE
ROBB Zigbee 8 button wall switch not pairing??

### DISCUSSION THREAD

**Discussion by bartve66** [Koenkk/zigbee2mqtt#9127] [Devices]
I bought the black version off the Robb 8 button wall switch (ROB_200-25-0), assuming it was just, well ... the black version of the supported device ROB_200-007-0.  I tried several times to pair it but the zigbee2mqtt log keeps reporting back 'Zigbee2MQTT:warn  2021-10-11 22:25:06: Device '0x60a423fffe65947d' with Zigbee model 'ROB_200-025-0' and manufacturer name 'ROBB smarrt' is NOT supported, please follow https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html'.
I find it hard to believe its a different device, apart from the color? Maybe the different device code is messing things up?

**Comment by bartve66** [comment]
A little bump (sorry) and some additional findings: I tried to add the witch as an new device according to the howto (How to support new devices). the first step of the howto says: 'It should be possible to pair your unsupported device out of the box because Zigbee2MQTT can pair with any zigbee device.'
This is were things already go of the rails because after (attempting) to pair the switch the z2m log says: 'Zigbee2MQTT:warn  2021-10-15 13:23:32: Received message from unsupported device with Zigbee model 'undefined' and manufacturer name 'undefined''
So pairing is not possible?
I read in the howto I could start an issue for this, I would be happy to go through proper channels for addressing this?

**Comment by bartve66** [comment]
I just keep trying ...  I updated the firmware of my zzh stick, and that resulted that in the warning 'Received message from ...  with Zigbee model 'undefined' ...' now '... model 'undefined'' is replaced with ' ... model 'ROB_200-025-0'', but still pairing was not succesfull.
Next I used a longer USB extension cable during pairing, and now pairing worked!

But still a unsupported device ...

Now I see on github in zigbee-herdsman-converters/devices/robb.js the converter (?) for the white variant of the switch:
```
    {
        zigbeeModel: ['ZG2833K8_EU05', 'ROB_200-007-0'],
        model: 'ROB_200-007-0',
        vendor: 'ROBB',
        description: 'Zigbee 8 button wall switch',
        fromZigbee: [fz.command_on, fz.command_off, fz.command_move, fz.command_stop, fz.battery, fz.ignore_genOta],
        exposes: [e.battery(), e.action([
            'on_1', 'off_1', 'brightness_move_up_1', 'brightness_move_down_1', 'brightness_stop_1',
            'on_2', 'off_2', 'brightness_move_up_2', 'brightness_move_down_2', 'brightness_stop_2',
            'on_3', 'off_3', 'brightness_move_up_3', 'brightness_move_down_3', 'brightness_stop_3',
            'on_4', 'off_4', 'brightness_move_up_4', 'brightness_move_down_4', 'brightness_stop_4'])],
        toZigbee: [],
        meta: {multiEndpoint: true, battery: {dontDividePercentage: true}},
        whiteLabel: [{vendor: 'Sunricher', model: 'SR-ZG9001K8-DIM'}],
    },
```
Now am I wrong to think if a similar piece of code was added with 'ROB_200-007-0' replaced by 'ROB_200-025-0' the black variant of the switch would be supported by z2m?
I only really don't know how to do that: I am running Home Assistant on Home Assistant OS, and z2m as an addon in Home Assistant: I can edit z2m configuration.yaml via the Supervisor in the HA web interface, but can not (or don't know how to) access, add or edit  other z2m config files, like .js converter files.

**Reply by Koenkk** [reply]
that will likely work indeed, added it.

Changes will be available in the dev branch in a few hours from now. (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Comment by bartve66** [comment]
Great, thanx! 
I am now on the stable version of z2m (1.21.2-1), so a bit hard for me to test right now. I guess the current edge version will be the next stable version? Any idea when that will happen?

**Reply by Koenkk** [reply]
1 November

**Comment by bartve66** [comment]
Top! Thanx again!
